src/agroservices/ipm.py

```diff
@@ -218,18 +220,53 @@ def get_weatheradapter(
             parameters: (list)  Comma separated list of the requested weather parameters
             timeStart: Start of weather data period (ISO-8601 Timestamp, e.g. 2020-06-12T00:00:00+03:00)
             timeEnd: End of weather data period (ISO-8601 Timestamp, e.g. 2020-07-03T00:00:00+03:00)
-            weatherStationId: The weather station id (FMISID) 
-                              in the open data API https://en.ilmatieteenlaitos.fi/observation-stations?filterKey=groups&filterQuery=weather
+            weatherStationId: The weather station id (FMISID) in the open data API https://en.ilmatieteenlaitos.fi/observation-stations?filterKey=groups&filterQuery=weather
 
         Returns:
         --------
             weather observations in the IPM Decision's weather data format in json format
         """
-        # test endpoint argument
+
+        # Test
+        ############
+
+        sources = self.get_weatherdatasource()
+
+        ## test endpoint argument
         endpoints = self.weatheradapter_service(forecast=False)
         if not endpoint in endpoints.values():
             raise ValueError("endpoint error: weatheradapter service not exit \n"
                              "or is a forecast weatheradapter in this case used weatheradapter_forecast")
+
+        ## test credentials
+        authentification = {item["endpoint"].split("rest")[1]:item['authentication_required']for item in sources}
+        if authentification[endpoint]=='false':
+            credentials is None
```

Comment:
What does it mean?
credential = None ?
credentials is None is a test. It returns True or False
If you want to assign credential to None:
`credentials = None`

Reply:
is the test correct now?